project_task_materials_stock/models/project.py

```diff
@@ -0,0 +1,180 @@
+                                _("You can't move to a not consume stage if "
+                                  "there are already analytic lines")
+                            )
+                    task.material_ids.mapped('analytic_line_id').unlink()
+        return res
+
+    @api.multi
+    def unlink(self):
+        self.mapped('stock_move_ids').unlink()
+        self.mapped('analytic_line_ids').unlink()
+        return super(Task, self).unlink()
+
+    @api.multi
+    def action_assign(self):
+        self.mapped('stock_move_ids').action_assign()
+
+    @api.multi
+    def action_done(self):
+        self.mapped('stock_move_ids').action_done()
+
+
+class ProjectTaskMaterials(models.Model):
+    _inherit = "project.task.materials"
+
+    stock_move_id = fields.Many2one(
+        comodel_name='stock.move', string='Stock Move')
+    analytic_line_id = fields.Many2one(
+        comodel_name='account.analytic.line', string='Analytic Line')
+    product_uom = fields.Many2one(
+        comodel_name='product.uom', string='Unit of Measure')
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.product_uom = self.product_id.uom_id.id
+        return {'domain': {'product_uom': [
+            ('category_id', '=', self.product_id.uom_id.category_id.id)]}}
+
+    def _prepare_stock_move(self):
+        product = self.product_id
+        res = {
+            'product_id': product.id,
+            'name': product.partner_ref,
+            'state': 'confirmed',
+            'product_uom': self.product_uom.id or product.uom_id.id,
+            'product_uom_qty': self.quantity,
+            'origin': self.task_id.name,
+            'location_id': self.env.ref(
+                'stock.stock_location_stock').id,
+            'location_dest_id': self.env.ref(
+                'stock.stock_location_customers').id,
+        }
+        return res
+
+    @api.multi
+    def create_stock_move(self):
+        for line in self:
+            move_id = self.env['stock.move'].create(
+                line._prepare_stock_move())
+            line.stock_move_id = move_id.id
+
+    def _prepare_analytic_line(self):
+        product = self.product_id
+        company_id = self.env['res.company']._company_default_get(
+            'account.analytic.line')
+        analytic_account = getattr(self.task_id, 'analytic_account_id', False)\
+            or self.task_id.project_id.analytic_account_id
+        res = {
+            'name': self.task_id.name + ': ' + product.name,
+            'ref': self.task_id.name,
+            'product_id': product.id,
+            'unit_amount': self.quantity,
+            'account_id': analytic_account.id,
+            'user_id': self._uid,
+            'product_uom_id': self.product_uom.id,
+        }
+        ctx = dict(self._context or {})
+        ctx['uom'] = self.product_uom.id
+        # Compute based on pricetype
+        amount_unit = \
+            self.product_id.with_context(ctx).price_get(
```

Comment:
Better do `self.product_id.with_context(uom=self.product_uom.id).price_get(` and remove those `ctx` lines above.